pkg/ccl/backupccl/restore_job.go

```diff
@@ -1265,6 +1265,12 @@ func (r *restoreResumer) doResume(ctx context.Context, execCtx interface{}) erro
 		return err
 	}
 
+	// To maintain mixed version compatibility where we plan RESTORE SYSTEM USERS on 22.1
+	// (not aware of the tree.SystemUsers enum) but execute on 22.2.
```

Comment:
nit: wrap comments at 70 characters

another nit: the comment sounds incomplete imo maybe reword to:

// A `RESTORE SYSTEM USERS` planned on a 22.1 node will use the `RestoreSystemUsers` field
// in the job details to identify this flavour of RESTORE. We must continue to check this field
// for mixed-version compatability.
//
// TODO(msbutler): Delete in 23.1

Reply:
Done.